lightningd/pay.c

```diff
@@ -37,15 +37,16 @@ struct sendpay_command {
 	struct command *cmd;
 };
 
-static bool string_to_payment_status(const char *status_str, enum wallet_payment_status *status)
+static bool string_to_payment_status(const char *status_str, size_t len,
+				     enum wallet_payment_status *status)
 {
-	if (streq(status_str, "complete")) {
```

Comment:
Just out of curiosity: how much performance difference does this make to the comparison? I'd have expected this to be pretty close, maybe a bit less branchy due to the predictable length in `memeqstr`

Reply:
Yes, when we're parsing JSON we don't have a nul-terminated string :(